cluster-autoscaler/processors/provreq/provisioning_request_processors_test.go

```diff
@@ -0,0 +1,58 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package provreq
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	apiv1 "k8s.io/api/core/v1"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/autoscaler/cluster-autoscaler/context"
+	. "k8s.io/autoscaler/cluster-autoscaler/utils/test"
+	"k8s.io/client-go/tools/record"
+)
+
+func TestProvisioningRequestPodsFilter(t *testing.T) {
+	prPod := BuildTestPod("pr-pod", 500, 10)
+	prPod.Annotations[provisioningRequestPodAnnotationKey] = "provisioning-request-class"
+	pod := BuildTestPod("pod", 500, 10)
+	testCases := map[string]struct {
+		unschedulableCandidates []*apiv1.Pod
+		expectedUnscheduledPods []*apiv1.Pod
+	}{
+		"ProvisioningRequest consumer is filtered out": {
+			unschedulableCandidates: []*v1.Pod{prPod, pod},
+			expectedUnscheduledPods: []*v1.Pod{pod},
+		},
+	}
+	for _, test := range testCases {
+		eventRecorder := record.NewFakeRecorder(10)
+		ctx := &context.AutoscalingContext{AutoscalingKubeClients: context.AutoscalingKubeClients{Recorder: eventRecorder}}
+		filter := NewProvisioningRequestPodsFilter(NewDefautlEventManager())
+		got, _ := filter.Process(ctx, test.unschedulableCandidates)
+		assert.ElementsMatch(t, got, test.expectedUnscheduledPods)
+		select {
+		case event := <-eventRecorder.Events:
+			assert.Contains(t, event, "Unschedulable pod ignored in scale-up loop")
```

Comment:
`Unschedulable pod ignored in scale-up loop, because it's consuming ProvisioningRequest $NAMESPACE/provisioning-request-class`